language/diem-tools/writeset-transaction-generator/src/release.rs

```diff
@@ -71,3 +81,95 @@ pub fn create_release_writeset(
         vec![],
     )))
 }
+
+pub fn verify_payload_change<I: DiemValidatorInterface>(
+    validator: &I,
+    block_height_opt: Option<Version>,
```

Comment:
Maybe get rid of the option here? It's not used at any call site, and ideally this should always return the same result given the same parameters.

Reply:
Ah! There's a specific reason why I would like to make it optional. The reason is I want this function to be invoked, it would be nice if we just run these checks against the latest state known.

Reply:
I understand that, but I'm not sure it's a good idea in this case. It's definitely annoying to have to enter the version manually, but I think in this case the friction might be a feature. Verification is always relative to a version, and the version is always moving upward. If we try to hide this from the user, it might lead to incorrect assumptions and/or hard-to-debug failures.

For example: imagine the Move oncall creates a DF release with`--latest`. AOS grabs the bytes from the release and attempts to re-validate the write set before signing it, but now validation fails. Because there's no record of the version the oncall and AOS validated against, it will be very hard to determine what went wrong. By contrast, if both know the versions, it is easy to bisect and find out where validation started failing. 

If we force the user to enter the version manually, it should always be possible to do this.